### PR TITLE
[16.0][ADD] stock_picking_report_qty_undelivered

### PR DESCRIPTION
**Pull request by EmilioPascual**
This module was developed because only products that have not yet been delivered are printed on the delivery slip, i.e. products included in backorder.

With this module you can print a summary of the products that will not be delivered even if the backorder is not created.

https://www.loom.com/share/829350705bb5408985fa2bed9755f7e4

@Shide @Gelojr @rafaelbn  please review.

MT-5244 @moduon

### CODE REVIEW

**Review by fcvalgar — APPROVED** [review]
Functional review,

LGTM, thank you @EmilioPascual 

image

image

File: stock_picking_report_qty_undelivered/views/stock_picking_type_views.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<!-- Copyright 2024 Moduon Team S.L.
+     License GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0) -->
+<odoo>
+    <record id="view_picking_type_form_inherit_qty_not_done" model="ir.ui.view">
+        <field name="name">Stock Picking Type Qty Not Done</field>
+        <field name="model">stock.picking.type</field>
+        <field name="inherit_id" ref="stock.view_picking_type_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='create_backorder']/parent::group"
```
Comment (Shide) [inline stock_picking_report_qty_undelivered/views/stock_picking_type_views.xml]: **nitpick:** Simpler

```suggestion
                expr="//field[@name='create_backorder']/.."
```

**Review by Shide — APPROVED** [review]
Code LGTM

**Review by rafaelbn — APPROVED** [review]
Tested in prod 👍🏼 

/ocabot merge patch

**Pull request merged**